File: superstitioApi/java/superstitioapi/pet/animationSize/AnimationSize.java
package superstitioapi.pet.animationSize;

import basemod.ReflectionHacks;
import com.badlogic.gdx.math.MathUtils;
import com.esotericsoftware.spine.AnimationState;
import com.megacrit.cardcrawl.core.AbstractCreature;

public class AnimationSize {

    public static void reloadAnimation(AbstractCreature creature, float scale) {
        if (!AnimationUrlData.urlDataMap.containsKey(creature.getClass().getName())) return;
        AnimationState.TrackEntry oldTrackEntry = creature.state.getTracks().first();
        if (oldTrackEntry == null) return;
        AnimationUrlData urlData = AnimationUrlData.urlDataMap.get(creature.getClass().getName());
        ReflectionHacks.privateMethod(AbstractCreature.class, "loadAnimation", String.class, String.class, float.class)
                .invoke(creature, urlData.atlasUrl, urlData.skeletonUrl, scale);
        AnimationState.TrackEntry trackEntry;
        trackEntry = creature.state.setAnimation(0, oldTrackEntry.getAnimation().getName(), oldTrackEntry.getLoop());
        AnimationStateDataParam.tryReload(creature);
        trackEntry.setTime(trackEntry.getEndTime() * MathUtils.random());
    }
}

File: superstitioApi/java/superstitioapi/pet/animationSize/AnimationSize.kt
package superstitioapi.pet.animationSize

import basemod.ReflectionHacks
import com.badlogic.gdx.math.MathUtils
import com.megacrit.cardcrawl.core.AbstractCreature

object AnimationSize
{
    fun reloadAnimation(creature: AbstractCreature, scale: Float)
    {
        if (!AnimationUrlData.urlDataMap.containsKey(creature.javaClass.name)) return
        val oldTrackEntry = creature.state.tracks.first() ?: return
        val urlData: AnimationUrlData = AnimationUrlData.urlDataMap.get(creature.javaClass.name)!!
        ReflectionHacks.privateMethod(
            AbstractCreature::class.java,
            "loadAnimation",
            String::class.java,
            String::class.java,
            Float::class.javaPrimitiveType
        )
            .invoke<Any>(creature, urlData.atlasUrl, urlData.skeletonUrl, scale)
        val trackEntry = creature.state.setAnimation(0, oldTrackEntry.animation.name, oldTrackEntry.loop)
        AnimationStateDataParam.tryReload(creature)
        trackEntry.time = trackEntry.endTime * MathUtils.random()
    }
}
